### PR TITLE
Add default values to `calc`

### PR DESCRIPTION
**Pull request by browner12**
this will allow the user to only specify the width, if they are doing an n:1 aspect ratio.

```html
<div class="aspect-w-1 aspect-h-1">...</div>
```

can be simplified to:

```html
<div class="aspect-w-1">...</div>
```

### CODE REVIEW

File: src/index.js
Comment (RobinMalfait) [inline src/index.js:24]:
```suggestion
            paddingBottom: `calc(var(--tw-aspect-h, var(--tw-aspect-w)) / var(--tw-aspect-w) * 100%)`,
```

**Review by RobinMalfait — CHANGES_REQUESTED** [review]
Hey! Thank you for your PR!
Much appreciated! 🙏

We have a few shorthands in Tailwind itself.
However, this also results in something odd like this:

`aspect-w-10` will result in an aspect ratio of `10` by `1`, while it looks weird I think that I would expect a `10` by `10` which results in a square of `1` by `1`.

To accomplish this, we could rewrite it a little bit. I also think that the default value for `--tw-aspect-w` is unnecessary since it will always be set once you apply a class of `aspect-w-*`.

What do you think?

**Comment by browner12** [comment]
If I'm understanding you correctly, then every declaration of solely `.aspect-w-*` would result in a square.

While I imagine a 'square' would be the most common use case for only setting one value, I don't see any reason to not leave the option open, if someone wanted something like a 2:1 or 3:1.

In my mind, "1" seems like the natural default number here, so `.aspect-w-10` being 10:1 makes the most sense in my head.

IMO, it is nice to leave default values for both variables, to avoid undefined values, but you're right, technically we could drop the default for `--tw-aspect-w` if we are required to declare a `.aspect-w-*` class at all times.

---

Does the Tailwind team have a policy regarding wildcard selectors? Something really cool we could do is allow the user to set *either* `.aspect-w-*` *or* `.aspect-h-*`, and have the other default to a "1".

Back of the napkin CSS might look something like this:

```css
[class^="aspect-w-"], [class^="aspect-h-"] {
  position: 'relative',
  paddingBottom: `calc(var(--tw-aspect-h) / var(--tw-aspect-w) * 100%)`,
},

[class^="aspect-w-"], [class^="aspect-h-"]: {
  position: 'absolute',
  height: '100%',
  width: '100%',
  top: '0',
  right: '0',
  bottom: '0',
  left: '0',
},
```

**Comment by RobinMalfait** [comment]
Yeah that would mean everything is going to be square indeed. I'm just not 100% convinced that `10:1` is the correct "default".

Regarding the `aspect-w-*`, I can see a future where we allow something like that in the future once this project launches: https://twitter.com/adamwathan/status/1370143406337490950?s=20

**Comment by browner12** [comment]
Either option is better than what we currently have, so I don't think we can go wrong here.

Don't focus on the `10:1`, since it's highly unlikely someone would ever use that. Think about the `2:1` or `3:1` which are more realistic values. If you leave my changes in place, the user could simply write `<div class="aspect-w-2">` or `<div class="aspect-w-3">`.  If we make your adjustments, both of those options would result in the same thing, a square.

I'd vote for leaving as is, but I'll leave it up to you as the maintainer to make the final call.

**Comment by RobinMalfait** [comment]
Sorry for the long wait. I think for now we want to keep the plugin as-is, yes it is verbose but it is explicit and results in the expected behaviour.

This plugin always has been a temporary solution and now that we have native aspect-ratio support in most browsers (and also in Tailwind itself) I don't think this change is necessary.

The new internal plugin behaves more like how you would expect it to behave where you typically define a single name and have the aspect-ratio ready instead of using 2 values. https://tailwindcss.com/docs/aspect-ratio

**Pull request closed without merging**
